core/src/main/java/com/linecorp/armeria/client/proxy/ProxyConfig.java

```diff
@@ -0,0 +1,112 @@
+    }
+
+    /**
+     * Creates a {@code ProxyConfig} configuration for SOCKS4 protocol.
+     * @param proxyAddress The proxy address.
+     * @param username The username.
+     */
+    public static Socks4ProxyConfig socks4(InetSocketAddress proxyAddress, String username) {
+        return new Socks4ProxyConfig(requireNonNull(proxyAddress, "proxyAddress"),
+                                     requireNonNull(username, "username"));
+    }
+
+    /**
+     * Creates a {@code ProxyConfig} configuration for SOCKS5 protocol.
+     * @param proxyAddress The proxy address.
+     */
+    public static Socks5ProxyConfig socks5(InetSocketAddress proxyAddress) {
+        return new Socks5ProxyConfig(requireNonNull(proxyAddress, "proxyAddress"), null, null);
+    }
+
+    /**
+     * Creates a {@code ProxyConfig} configuration for SOCKS5 protocol.
+     * @param proxyAddress The proxy address.
+     * @param username The username.
+     * @param password The password.
+     */
+    public static Socks5ProxyConfig socks5(
+            InetSocketAddress proxyAddress, String username, String password) {
+        return new Socks5ProxyConfig(
+                requireNonNull(proxyAddress, "proxyAddress"), requireNonNull(username, "username"),
+                requireNonNull(password, "password"));
+    }
+
+    /**
+     * Creates a {@code ProxyConfig} configuration for CONNECT protocol.
+     * @param proxyAddress The proxy address.
+     */
+    public static ConnectProxyConfig connect(InetSocketAddress proxyAddress) {
+        return new ConnectProxyConfig(requireNonNull(proxyAddress, "proxyAddress"), null, null, false);
+    }
+
+    /**
+     * Creates a {@code ProxyConfig} configuration for CONNECT protocol.
+     * @param proxyAddress The proxy address.
+     * @param useTls Whether to use TLS to connect to the proxy.
+     */
+    public static ConnectProxyConfig connect(
+            InetSocketAddress proxyAddress, boolean useTls) {
+        return new ConnectProxyConfig(
+                requireNonNull(proxyAddress, "proxyAddress"), null, null, useTls);
+    }
+
+    /**
+     * Creates a {@code ProxyConfig} configuration for CONNECT protocol.
+     * @param proxyAddress The proxy address.
+     * @param username The username.
+     * @param password The password.
+     * @param useTls Whether to use TLS to connect to the proxy.
+     */
+    public static ConnectProxyConfig connect(
+            InetSocketAddress proxyAddress, String username, String password, boolean useTls) {
+        return new ConnectProxyConfig(
+                requireNonNull(proxyAddress, "proxyAddress"), requireNonNull(username, "username"),
+                requireNonNull(password, "password"), useTls);
+    }
+
+    /**
+     * Returns a {@code ProxyConfig} which signifies that proxy is disabled.
+     */
+    public static ProxyConfig disabled() {
+        return DISABLED_PROXY_CONFIG;
+    }
+
+    ProxyConfig() {
```

Comment:
This class doesn't have any implementation. How about making this as an interface?

Reply:
It's because we don't want a user to extend it.

Reply:
Ah, thanks!